src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -548,6 +557,12 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             catch (Exception e)
             {
                 response?.Dispose();
+
+                if (e is OperationCanceledException operationException && TimeoutFired(callerToken, timeoutTime))
+                {
+                    throw CreateTimeoutException(operationException);
```

Comment:
HandleFinishSendAsyncError was doing some logging.  Do we need to do any in this path, too?

Reply:
Added HandleSendAsyncTimeout method logging the original exception and "Canceled due to timeout" message.